tradeexecutor/analysis/trade_analyser.py

```diff
@@ -451,14 +474,34 @@ def calculate_summary_statistics(self) -> TradeSummary:
 
             if position.is_win():
                 won += 1
+                average_winning_trade.append(position.realised_profit_percent)
+                average_winning_trade_duration.append(position.duration)
+
             elif position.is_lose():
                 lost += 1
+                average_losing_trade.append(position.realised_profit_percent)
+                average_losing_trade_duration.append(position.duration)
+
             else:
                 # Any profit exactly balances out loss in slippage and commission
                 zero_loss += 1
 
             profit += position.realised_profit
 
+        average_winning_trade_profit_pc = np.mean(average_winning_trade)
+        average_losing_trade_loss_pc = np.mean(average_losing_trade)
```

Comment:
Sanity check - do we really want to simply average the raw percentage values, or would it be better to use a weighted average taking trade sizes into account?

For example, if I risk 1 USD to buy an asset and then sell it for double the price (for 2 USD), II made a profit of 100%. Then in the next trade I risk 1000 USD and close the position for 1010 USD, realizing a modest profit of 1%.

According to the calculation, I made +55.5% per trade on average, but is this really the figure we want to report?
Because in reality I only made close to 1 with trading, because that 1000 USD trade dominates the other micro trade.

Perhaps a weighted average would be more informative?

Reply:
I agree with you, a weighted average would be more informative.

Reply:
Just please make sure with @miohtama that this is indeed what is required by the spec. Perhaps the traders are used to a plain average instead of a weighted one.